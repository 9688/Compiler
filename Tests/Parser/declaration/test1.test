int x = 5;

int f(){
	return x + 4;
}

void up(){
	x += 4;
}

void main(){
	struct {} *(*Z)[];
	int x = 0;
	x = f();
	up();
	return x;
}