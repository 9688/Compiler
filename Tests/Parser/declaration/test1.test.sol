<<<table #1>>>
<tags>
<types>
int
char
double
void
<vars>
var printf type function printf(var  type  to char, ){
} returned void
var scanf type function scanf(var  type  to char, ){
} returned void
var getchar type function getchar(){
} returned int
var x type int
ÀÄÄ{=}
   ÃÄÄ{x}
   ÀÄÄ{5}

var f type function (){
              <<<table #3>>>
              <tags>
              <types>
              <vars>
              ÀÄÄ{block}
                 ÀÄÄ{return}
                    ÀÄÄ{+}
                       ÃÄÄ{x}
                       ÀÄÄ{4}

} returned int
var up type function (){
               <<<table #5>>>
               <tags>
               <types>
               <vars>
               ÀÄÄ{block}
                  ÀÄÄ{=}
                     ÃÄÄ{x}
                     ÀÄÄ{+}
                        ÃÄÄ{x}
                        ÀÄÄ{4}

} returned void
var main type function (){
                 <<<table #7>>>
                 <tags>
                 <types>
                 <vars>
                 var Z type pointer to array size 20 of pointer to  record
                                                                      <<<table #8>>>
                                                                      <tags>
                                                                      <types>
                                                                      <vars>
                                                                      var x type int
                                                                    endrecord
                 var x type int
                 ÀÄÄ{=}
                    ÃÄÄ{x}
                    ÀÄÄ{0}

                 ÀÄÄ{block}
                    ÃÄÄ{=}
                    ³  ÃÄÄ{x}
                    ³  ÀÄÄ{call}
                    ³     ÀÄÄ{f}
                    ÃÄÄ{call}
                    ³  ÀÄÄ{up}
                    ÀÄÄ{empty expr}

} returned void
‘¥¬ ­â¨ç¥áª ï ®è¨¡ª  ¢ áâà®ª¥ 16, ¯®§¨æ¨¨ 10: äã­ªæ¨ï ­¥ ¤®«¦­  ¢®§¢à é âì §­ ç¥­¨¥

