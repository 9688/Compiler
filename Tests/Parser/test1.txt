int f1(void){
}

int f2(void* x){
}

int f3(int x, void* y){
}