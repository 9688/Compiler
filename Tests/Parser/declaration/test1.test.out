<--------TABLES-------->
<---TABLE--->
VARS::

x   int   
 = 5


f   FUNC () { BLOCK     RETURN
          "+"
               x
               4 } RETURNED int   


up   FUNC () { BLOCK
     "+="
          x
          4 } RETURNED void   


main   FUNC () { BLOCK
     "="
          x
          CALL
               f     CALL
          up     RETURN          x } RETURNED void   


TYPES::
char char

double double

int int

void void

<---TABLE--->
VARS::

TYPES::
<---TABLE--->
VARS::

TYPES::
<---TABLE--->
VARS::

Z   POINTER TO ARRAY () OF POINTER TO STRUCT $UNNAMED${
vars::

types::
}   


x   int   
 = 0


TYPES::
